src/DetailsView/components/failure-instance-panel-control.tsx

```diff
@@ -115,10 +130,73 @@ export class FailureInstancePanelControl extends React.Component<FailureInstance
         );
     }
 
+    protected getSelectorAndSnippet = (): JSX.Element => {
+        return (
+            <div>
+                <a className="learn-more"> Learn more about adding failure instances </a>
+                <TextField
+                    className="selector-failure-textfield"
+                    label="CSS Selector"
+                    multiline={true}
+                    rows={8}
+                    value={this.state.selector}
+                    onChange={this.onSelectorChange}
+                    resizable={false}
+                    defaultValue="CSS selector"
+                />
+                <div>Note: If the CSS selector maps to multiple snippets, the first will be selected.</div>
+                <div>
+                    <DefaultButton text="Validate CSS selector" onClick={this.onValidateSelector} disabled={this.state.selector === ''} />
+                </div>
+                <div>
+                    <label>Code Snippet</label>
+                    <div>{this.state.snippet}</div>
+                </div>
+            </div>
+        );
+    };
+
+    protected panelWithoutSelectorAndSnippet = (testStepConfig: Readonly<Requirement>, panelProps: GenericPanelProps): JSX.Element => {
+        return (
+            <GenericPanel {...panelProps}>
+                {testStepConfig.addFailureInstruction}
+                <TextField
+                    className="observed-failure-textfield"
+                    label="Observed failure"
+                    multiline={true}
+                    rows={8}
+                    value={this.state.failureDescription}
+                    onChange={this.onFailureDescriptionChange}
+                    resizable={false}
+                />
+                <ActionAndCancelButtonsComponent
+                    isHidden={false}
+                    primaryButtonDisabled={this.state.failureDescription === ''}
+                    primaryButtonText={this.props.actionType === CapturedInstanceActionType.CREATE ? 'Add' : 'Save'}
+                    primaryButtonOnClick={
+                        this.props.actionType === CapturedInstanceActionType.CREATE
+                            ? this.onAddFailureInstance
+                            : this.onSaveEditedFailureInstance
+                    }
+                    cancelButtonOnClick={this.closeFailureInstancePanel}
+                />
+            </GenericPanel>
+        );
+    };
+
     protected onFailureDescriptionChange = (event, value: string): void => {
         this.setState({ failureDescription: value });
     };
 
+    protected onSelectorChange = (event, value: string): void => {
```

Comment:
I see only one reference to this function (in this same class). Any special reason we need it to be `protected` instead of `private`?
Also, it looks like we're not testing this function. You may need to ask @karanbirsingh for more guidance on how to test this.